### PR TITLE
admin activity log search: trailing spaces are not trimmed #4512

### LINKED ISSUE #4512: admin activity log search: trailing spaces are not trimmed
Both `info:abc` and `info:abc[space]` should return the same results (irrespective of trailing spaces), but they don't.

### PR DESCRIPTION
**Pull request by chen-zhuo**
Fixes #4512.

Added a for-loop to trim each token of the query string.

### CODE REVIEW

File: src/main/java/teammates/ui/controller/AdminActivityLogPageData.java [hunk 1]
```diff
@@ -270,7 +270,10 @@ private QueryParameters parseQuery(String query) throws Exception{
             }
             
             String[] values = pair[1].split(",", -1);
-            String label = pair[0];
+            for (int j = 0; j < values.length; j++) {
+                values[j] = values[j].trim();
+            }
+            String label = pair[0].trim();
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageData.java]: Trimming can be extracted into a utility method (in `StringHelper`?)

**Comment by kanghj** [comment]
@chen-zhuo, the pr name is wrong. It should be the name of the issue.

**Comment by chen-zhuo** [comment]
@kanghj, I have edited this pr title.

@damithc, I am not sure what do you mean exactly. I can think of three ways of solving this issue. Tell me which one is preferred.

1. At around line 272, add a for-loop to make sure each member of the 'values' array gets trimmed. Do the same thing for 'label'. **(What I already did.)**
2. At around line 258, change the code from `query = query.replaceAll(" and ", "|");` to `query = query.replaceAll("\\s*?and\\s*?", "|");`. **(Is this better?)**
3. Write a method in 'StringHelper', which takes in an array of String and returns the array which each String trimmed. (Is this what you meant, @damithc?)

**Comment by damithc** [comment]
I meant option 3 :-)

**Comment by thyageshm** [comment]
@chen-zhuo could we try and complete this one by this release?

**Comment by thyageshm** [comment]
Closing this pull request due to inactivity

**Pull request closed without merging**